core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -906,7 +906,15 @@ class KafkaApis(val requestChannel: RequestChannel,
       val results = offsetFetchRequest.partitions.asScala.map { topicPartition => (topicPartition, unauthorizedGroupResponse)}.toMap
       new OffsetFetchResponse(results.asJava)
     } else {
-      val (authorizedTopicPartitions, unauthorizedTopicPartitions) = offsetFetchRequest.partitions.asScala.partition { topicPartition =>
+      val partitions =
+        if (header.apiVersion == 2 && offsetFetchRequest.isAllPartitions()) {
+          val group = coordinator.groupManager.getGroup(offsetFetchRequest.groupId)
```

Comment:
Sure. I think I'll expose a new method (something like `GroupCoordinator.getPartitions(groupId)`), since the partitions have to go through authorization check and, after a quick look, I don't see a clean way of overloading `handleFetchOffsets` for this purpose that fits well with how `KafkaApis.handleOffsetFetchRequest` is implemented. Unless we want to refactor that method more extensively.